pkg/redirectpolicy/manager.go

```diff
@@ -672,22 +662,43 @@ func (rpm *Manager) processConfigWithNamedPorts(config *LRPConfig, pods ...*podM
 				}
 			}
 			if len(bes4) > 0 {
-				rpm.upsertServiceWithBackends(config, feM, pod.id, bes4)
+				rpm.upsertFrontendMapping(config, feM, pod.id, bes4)
 			} else if len(bes6) > 0 {
-				rpm.upsertServiceWithBackends(config, feM, pod.id, bes6)
+				rpm.upsertFrontendMapping(config, feM, pod.id, bes6)
 			}
 		}
 	}
 }
 
-// upsertServiceWithBackends updates policy config internal state and upserts
-// service with the given pod backends.
-func (rpm *Manager) upsertServiceWithBackends(config *LRPConfig, frontendMapping *feMapping, podID podID, backends []backend) {
-	frontendMapping.backends = backends
-	rpm.policyPods[podID] = append(rpm.policyPods[podID], podPolicyInfo{
-		policyID: config.id,
-		backends: backends,
-	})
+// upsertFrontendMapping updates policy config internal state and upserts
+// service with the policy frontend and mapped backends.
+func (rpm *Manager) upsertFrontendMapping(config *LRPConfig, frontendMapping *feMapping, podID podID, backends []backend) {
+	newFePods := make([]backend, 0, len(frontendMapping.podBackends))
+	for _, be := range frontendMapping.podBackends {
+		if be.podID != podID {
+			newFePods = append(newFePods, be)
+		}
```

Comment:
can you add a comment describing what this logic is doing? It's not clear to me.